### PR TITLE
upload-assets-to-github-release 1.0.1

### PR DESCRIPTION
**Pull request by jasperroel**
TagCheck

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

This is a result of the feedback on https://github.com/bitrise-io/bitrise-steplib/pull/2335 (adding the `apt` dependencies in).

Note: there is no icon for this plugin.
# Testing
For testing, you need:
-  a valid "Github Personal Action Token" with `repo` permissions
- a valid "README.md" with content in your _tmp folder
- a valid repository/release to attach this file to

### CODE REVIEW

File: steps/upload-assets-to-github-release/1.0.1/step.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  opts:
+    description: |
+      Enables "hub" debug output
+    is_expand: true
+    is_required: false
+    summary: |
+      Enables "hub" debug output
+    title: Debug this step
+    value_options:
+    - "true"
```
Comment (lpusok) [inline steps/upload-assets-to-github-release/1.0.1/step.yml]: It is recommended to use "yes" and "no" as options for a boolean value.

File: steps/upload-assets-to-github-release/1.0.1/step.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - "false"
+- github_pat: null
```
Comment (lpusok) [inline steps/upload-assets-to-github-release/1.0.1/step.yml]: Could name this input `personal_access_token` no need to include github in the name as it is in the title.

File: steps/upload-assets-to-github-release/1.0.1/step.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    description: |
+      The Github Release to attach the files to. Should be the URL slug name (for example: `v3.15.0-21419`).
+      Defaults to the tag currently building.
+    is_expand: true
+    is_required: true
+    summary: |
+      The Github Release to attach the files to. Should be the URL slug name (for example: `v3.15.0-21419`).
+    title: Github Release name
+    value_options: []
+- files_to_attach_list: $BITRISE_APK_PATH_LIST|$BITRISE_IPA_PATH
```
Comment (lpusok) [inline steps/upload-assets-to-github-release/1.0.1/step.yml]: Could add BITRISE_AAB_PATH_LIST to include android app bundles.

File: steps/upload-assets-to-github-release/1.0.1/step.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+- osx-10.10
+- ubuntu-16.04
+type_tags:
+- deploy
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: git
```
Comment (lpusok) [inline steps/upload-assets-to-github-release/1.0.1/step.yml]: Please remove git if not used from the step source code.

**Comment by lpusok** [comment]
Thanks for the contribution @jasperroel !
Please share the step again with the fixes.

**Comment by jasperroel** [comment]
@lpusok Thanks for the suggestion! I'm building a new version with some new features now.

Another thing that came up is the "hub" requirement, maybe you can help me with that.
In `osx-10.10`, that package is available via brew, so no problem there.

But brew is only used on MacOS. For the ubuntu OS (`ubuntu-16.04`), that is actually a (stripped down) docker version. There is no apt version of this package to install, but Github wants to use "snap" (https://github.com/github/hub -> `snap install hub --classic`). Would you know if that's available?

Alternativally, I can experiment with bundling the executable in this repo for Ubuntu.

**Comment by trapacska** [comment]
> Would you know if that's available?

I am afraid that is not possible through our brew/apt-get installer then. You can prepare your step's logic to prepare the environment and install missing dependencies.

Closing this issue as re-share will be required.
Thanks!

**Pull request closed without merging**